src/fragments/lib/auth/flutter/getting_started/40_authenticator.mdx

```diff
@@ -0,0 +1,260 @@
+      ),
+      SignUpFormField.password(),
+      SignUpFormField.passwordConfirmation(),
+    ],
+  ),
+  child: const Scaffold(
+    body: Center(child: Text('You are logged in!')),
+  ),
+);
+```
+
+To add a custom form field that is not available in the pre-defined list provided by Cognito:
+
+```dart
+final authenticator = Authenticator(
+  signUpForm: SignUpForm.custom(
+    fields: [
+      SignUpFormField.username(
+        validator: _validateUsername,
+      ),
+      //Example custom field
+      SignUpFormField.custom(
+          title: 'Favorite book',
+          attributeKey:
+              const CognitoUserAttributeKey.custom('favoritebook'),
+          hintText: 'Enter your favorite book'),
+      SignUpFormField.password(),
+      SignUpFormField.passwordConfirmation(),
+    ],
+  ),
+  child: const Scaffold(
+    body: Center(child: Text('You are logged in!')),
+  ),
+);
+
+String? _validateUsername(UsernameInput? input) {
+  final username = input?.username;
+  if (username == null || username.isEmpty) {
+    return 'Username cannot be empty';
+  }
+
+  bool containsAmplify = username.contains('amplify');
+  if (!containsAmplify) {
+    return 'Username needs to include amplify';
+  }
+
+  return null;
+}
+```
+
+### Text overrides and internationalization
+
+The text displayed in the Authenticator is customizable and compatible with Flutter's out-of-the-box internationalization APIs.
+
+#### Simple text customization
+
+It is not necessary to do full fledged localization to customize components within the Authenticator. Below is a simple demonstration:
+
+```dart
+class LocalizedButtonResolver extends ButtonResolver {
+  const LocalizedButtonResolver();
+
+  @override
+  String signin(BuildContext context) {
+    //the text of the signin button can be customized
+    return 'Login to Amplify';
+  }
+}
+
+class LocalizedInputResolver extends InputResolver {
+  const LocalizedInputResolver();
+
+  @override
+  String title(BuildContext context, InputField field) {
+    switch (field) {
+      case InputField.username:
+        return 'Custom Username Title';
+      case InputField.email:
+        return 'Custom Email Title';
+      case InputField.password:
+        return 'Custom Password Title';
+      case InputField.passwordConfirmation:
+        return 'Custom Password Confirmation Title';
+      default:
+        return super.title(context, field);
+    }
+  }
+}
+
+
+const stringResolver = AuthStringResolver(
+  buttons: LocalizedButtonResolver(),
+);
+
+final authenticator = Authenticator(
+  stringResolver: stringResolver,
+  child: const Text('You are logged in!'),
+);
+```
+
+#### Setting up internationalization
+
+
+To use true localiztion, you should carefully follow the steps in the [Flutter Localization guide](https://flutter.dev/docs/development/accessibility-and-localization/internationalization). When the guide says to create an .arb file, enter the following into `lib/l10n/amplify_en.arb`:
+
+```json
```

Comment:
We might want to link to the arb files in the package. Otherwise I am not sure how a customer will know what the other keys are. This isn't something I am actually that familiar with. Maybe @dnys1 or @haverchuck have a better idea on how we can make this easier.

Reply:
Not that familiar with it as well. That was one of the things I needed more help with in the quip doc.

Reply:
All of our defaults strings are pulled from[ source .arbs](https://github.com/aws-amplify/amplify-flutter/blob/release-candidate/packages/amplify_authenticator/lib/src/l10n/src/buttons/buttons_en.arb) that are committed to the repo. We can probably just reference those?

Reply:
Internalization has been removed from this PR.